am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -219,6 +250,52 @@ public FilteredResourceEnvelope filterResource(@NotBlank String userId,
             .build();
     }
 
+    /**
+     * Removes empty nodes from data.
+     *
+     * @param node JsonNode
+     */
+    private void removeEmptyNodeValues(JsonNode node) {
+        if (nonNull(node)) {
+            Iterator<JsonNode> it = node.iterator();
+            while (it.hasNext()) {
+                JsonNode child = it.next();
+                if (child.isObject() && child.isEmpty(null)) {
+                    it.remove();
+                } else {
+                    removeEmptyNodeValues(child);
+                }
+            }
+        }
+    }
+
+    private Map<JsonPointer, Set<Permission>> filterAttributePermission(
+        Map<JsonPointer, Set<Permission>> attributePermissions, Set<JsonPointer> visibleAttributes) {
+
```

Comment:
Can you add comments to the methods?

Reply:
we can add comments...But in previous code check-ins no javadocs added for private method.

Reply:
done